frontend/src/components/problem/ProblemDisplay.tsx

```diff
@@ -0,0 +1,235 @@
+import styled from 'styled-components';
+import { deleteProblem, Problem, ProblemIOType } from '../../api/Problem';
+import { LargeInputButton, TextInput } from '../core/Input';
+import Difficulty from '../../api/Difficulty';
+import { DifficultyButton, ProblemIOTypeButton, SmallButton } from '../core/Button';
+import { MediumText, Text } from '../core/Text';
+import Loading from '../core/Loading';
+import ErrorMessage from '../core/Error';
+
+const Content = styled.div`
+  padding: 10px;
+`;
+
+type ProblemDisplayParams = {
+  problem: Problem,
+  actionText: string,
+  onClick: (newProblem: Problem) => void,
+  editMode: boolean,
+};
+
+function ProblemDisplay(props: ProblemDisplayParams) {
+  const {
+    problem, onClick, actionText, editMode,
+  } = props;
+
+  const history = useHistory();
+  const [newProblem, setNewProblem] = useState<Problem>(problem);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+
+  const deleteProblemFunc = () => {
+    if (!window.confirm('Are you sure you want to delete this problem?')) {
+      return;
+    }
+
+    setLoading(true);
+    setError('');
+
+    deleteProblem(newProblem.problemId)
+      .then(() => {
+        setLoading(false);
+        history.replace('/problems/all');
+      })
+      .catch((err) => {
+        setError(err.message);
+        setLoading(false);
+      });
+  };
+
+  // Handle updating of normal text fields
+  const handleChange = (e: any) => {
+    const { name, value } = e.target;
+    setNewProblem({
+      ...newProblem,
+      [name]: value,
+    });
+  };
+
+  // Handle updating of enum-type fields
+  const handleEnumChange = (name: string, value: any) => handleChange({ target: { name, value } });
+
+  // Handle updating of problem inputs
+  const handleInputChange = (index: number, name: string, type: ProblemIOType) => {
+    setNewProblem({
+      ...newProblem,
+      problemInputs: newProblem.problemInputs.map((input, i) => {
+        if (index === i) {
+          return { name, type };
+        }
+        return input;
+      }),
+    });
+  };
+
+  // Handle adding a new problem input for this problem
+  const addProblemInput = () => {
+    setNewProblem({
+      ...newProblem,
+      problemInputs: [...newProblem.problemInputs, { name: 'name', type: ProblemIOType.Integer }],
+    });
+  };
+
+  // Handle deleting a problem input for this problem
+  const deleteProblemInput = (index: number) => {
+    setNewProblem({
+      ...newProblem,
+      problemInputs: newProblem.problemInputs.filter((_, i) => index !== i),
+    });
+  };
+
+  // Handle updating of test case
+  const handleTestCaseChange = (index: number, input: string, output: string) => {
+    setNewProblem({
+      ...newProblem,
+      testCases: newProblem.testCases.map((testCase, i) => {
+        if (index === i) {
+          return {
+            input,
+            output,
+            hidden: false,
+            explanation: '',
```

Comment:
Sounds good - I went against it at first since personally I do feel like those fields are kinda irrelevant (at least for the MVP) but it's always easier to delete than implement so I went ahead and added those too